crates/legacy/src/service.rs

```diff
@@ -416,13 +416,28 @@ impl<Types: NodeType> GlobalState<Types> {
         txn_status: TransactionStatus,
     ) -> Result<(), BuildError> {
         if self.tx_status.write_arc().await.contains(&txn_hash) {
+            let mut write_guard = self.tx_status.write_arc().await;
+            let old_status = write_guard.get(&txn_hash);
+            match old_status {
+                Some(TransactionStatus::Rejected { reason }) => {
+                    tracing::error!("Changing the status of a rejected transaction to status {:?} is not allowed! The reason it is rejected is {:?}", txn_status, reason);
```

Comment:
We will run into most of this cases quite frequently during normal operation, I'd downgrade this to debug.

Reply:
Why is that? Is this due to a rejected txn could be submitted twice?

Reply:
Yeah, I'd expect cases where we get duplicates either directly or through hotshot gossiping not to be very rare.
It doesn't indicate that something's wrong with the builder, so I wouldn't categorize this as an error.

Reply:
I see, then I'd better also not return BuildError in such cases.

Reply:
Updated in [5b3f1b7](https://github.com/EspressoSystems/marketplace-builder-core/pull/236/commits/5b3f1b74a7ee65e6cc4d09432e482489451f8723).